### PR TITLE
fix: preserve trailing optional path parameters

### LINKED ISSUE #2168: Optional path parameters do not match correctly
### Scope

Improves an existing behavior

### Compatibility

- [ ] This is a breaking change

### Feature description

There is a [step](https://github.com/mswjs/msw/blob/d97a528914df6a656dfe7b1f63985c85845df533/src/core/utils/url/cleanUrl.ts#L7-L12) that drops query parameters and it looks like it breaks the ability to specify optional URL parameters, e.g `/api/user/:id?`.

The cleanup routine should be limited to removing the query parameters only.

### PR DESCRIPTION
**Pull request by kettanaito**
- Fixes #2168

### CODE REVIEW

File: test/node/rest-api/request/matching/path-params-optional.node.test.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      HttpResponse.json({ mocked: true }),
+    ),
+  )
+
+  const response = await fetch('http://localhost/user/123')
+  expect(response.status).toBe(200)
+  expect(await response.json()).toEqual({ mocked: true })
+})
+
+it('intercepts the request that partially matches the path', async () => {
```
Comment (kettanaito) [inline test/node/rest-api/request/matching/path-params-optional.node.test.ts:33]: This test is currently failing. It's a good indication we have a bug for this scenario.

**Comment by KaiSpencer** [comment]
Hey @kettanaito,

I have a fix for this. How would you like it contributing given this branch and PR already existing.

**Comment by kettanaito** [comment]
Hi, @KaiSpencer. This should work:

1. Fork this repo.
2. Check out to this PR's branch.
3. From this branch, create a _new_ branch. Give it whichever name you wish.
4. Commit and push your changes to the new branch to _your fork_ as the origin.
5. Go on GitHub and open a PR from your new branch to _this_ feature branch. 

Then I'll see it and be able to review it!

**Comment by kettanaito** [comment]
## Released: v2.3.1 🎉

This has been released in v2.3.1!

- 📄 [**Release notes**](https://github.com/mswjs/msw/releases/tag/v2.3.1)
- 📦 [npm package](https://www.npmjs.com/package/msw/v/2.3.1)

Make sure to always update to the latest version (`npm i msw@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.

**Pull request merged**